spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -86,38 +86,148 @@
+        {
           1 => first_user.id,
           2 => second_user.id,
         }
+      end
+
+      let(:draft_revision) do
+        build(
+          :revision_history_event,
+          whodunnit: 1,
+          event: "create",
+          state: "draft",
+          created_at: 5.days.ago.noon,
+        )
+      end
+
+      let(:published_revision) do
+        build(
+          :revision_history_event,
+          whodunnit: 2,
+          event: "update",
+          state: "published",
+          created_at: 4.days.ago.noon,
+        )
+      end
+
+      let(:superseded_revision) do
+        build(
+          :revision_history_event,
+          whodunnit: 1,
+          event: "update",
+          state: "superseded",
+          created_at: 2.days.ago.noon,
+        )
+      end
 
-        whitehall_edition = build(
+      let(:editorial_remark) do
+        build(
+          :note_event,
+          author_id: 1,
+          body: "Another note",
+          state: "published",
+          created_at: 2.days.ago.noon,
+        )
+      end
+
+      let(:fact_check) do
+        build(
+          :fact_check_event,
+          requestor_id: 2,
+          email_address: "someone@somewhere.com",
+          comments: "Hello World",
+          state: "published",
+          created_at: 3.days.ago.noon,
+          updated_at: 1.day.ago.noon,
+        )
+      end
+
+      let(:whitehall_edition) do
+        build(
           :whitehall_export_edition,
-          revision_history: [
-            build(:revision_history_event, whodunnit: 1, event: "create", state: "draft", created_at: 2.days.ago),
-            build(:revision_history_event, whodunnit: 2, event: "update", state: "published", created_at: 1.day.ago),
-            build(:revision_history_event, whodunnit: 1, event: "update", state: "superseded", created_at: 2.days.ago),
-          ],
+          revision_history: [draft_revision, published_revision, superseded_revision],
+          editorial_remarks: [editorial_remark],
+          fact_check_requests: [fact_check],
+        )
+      end
+
+      let(:edition) do
+        described_class.call(
+          document_import: document_import,
+          whitehall_edition: whitehall_edition,
+          user_ids: user_ids,
         )
+      end
 
-        edition = described_class.call(document_import: document_import,
-                                       whitehall_edition: whitehall_edition,
-                                       user_ids: user_ids)
+      it "has a draft revision" do
+        timeline_entry = edition.timeline_entries.first
+        expect(timeline_entry.attributes).to match a_hash_including(
+          "entry_type" => "whitehall_migration",
+          "created_at" => draft_revision["created_at"],
+          "created_by_id" => first_user.id,
+        )
+        expect(timeline_entry.details.attributes).to match a_hash_including(
+          "entry_type" => "first_created",
+          "contents" => "{}",
+        )
+      end
```

Comment:
Argh I wrote a long comment here before but it seems lost in an abyss somewhere.

I found this test, and the other ones, somewhat confusing and inconsistent with the other tests in this file because the actual work for testing the code is done in the `let(:edition)` block. Whereas other tests in this file (and mostly across the app) try to focus on executing the code under test in the test itself. 

I think these tests can be improved by doing more in the test itself. I think this would have benefits in making the tests isolated (at the moment you need to read a lot of code to understand why say an editiorial remark mark looks up the third timeline entry of an edition) and reduce the amount of lets to go through which can be a bit of a smell (see: https://thoughtbot.com/blog/lets-not)

As an example:

```
it "imports a create revision history as a first_created event" do
  create_revision_event = build(:revision_history_event,
                                whodunnit: 1,
                                event: "create",
                                state: "draft",
                                created_at: 5.days.ago.noon)
  whitehall_edition = build(:whitehall_export_edition, revision_history: [create_revision])
  edition = described_class.call(document_import: document_import,
                                 whitehall_edition: whitehall_edition,
                                 user_ids: user_ids)
  timeline_entry = edition.timeline_entries.first
  expect(timeline_entry.attributes)
    .to match a_hash_including("entry_type" => "whitehall_migration",
                               "created_at" => 5.days.ago.noon,
                               "created_by_id" => first_user.id)
  expect(timeline_entry.details.attributes)
    .to match a_hash_including("entry_type" => "first_created",
                               "contents" => "{}")
end
```

There's a good chance this may increase lines of code slightly but I think it'll be beneficial to the overall code.

Reply:
Thanks for the suggestion and keeping me on track!  Tests updated.